management/clipper_manager.py

```diff
@@ -227,17 +227,21 @@ def _execute_root(self, *args, **kwargs):
         if not self.sudo:
             return self._execute_standard(*args, **kwargs)
         elif self._host_is_local():
-            return self._execute_local(*args, **kwargs)
+            return self._execute_local(True, *args, **kwargs)
         else:
             return sudo(*args, **kwargs)
 
     def _execute_standard(self, *args, **kwargs):
         if self._host_is_local():
-            return self._execute_local(*args, **kwargs)
+            return self._execute_local(False, *args, **kwargs)
         else:
             return run(*args, **kwargs)
 
-    def _execute_local(self, *args, **kwargs):
+    def _execute_local(self, as_root, *args, **kwargs):
+        if self.sudo and as_root:
+            root_args = list(args)
+            root_args[0] = "sudo %s" % root_args[0]
```

Comment:
The first argument is the entirety of the docker command. For example, we have 
```py
args[0] = "docker-compose up -d query_frontend"
```
when we call
```py
self._execute_root("docker-compose up -d query_frontend")
```